share/models/ingest.py

```diff
@@ -122,21 +125,34 @@ class SourceConfig(models.Model):
     def natural_key(self):
         return (self.label,)
 
-    def get_harvester(self):
-        return self.harvester.get_class()(self, **(self.harvester_kwargs or {}))
+    def get_harvester(self, **kwargs):
+        return self.harvester.get_class()(self, **{**kwargs, **(self.harvester_kwargs or {})})
 
     def get_transformer(self):
         return self.transformer.get_class()(self, **(self.transformer_kwargs or {}))
 
-    def acquire_lock(self, using='locking'):
+    @contextlib.contextmanager
+    def acquire_lock(self, required=True, using='default'):
+        from share.harvest.exceptions import HarvesterConcurrencyError
+
         # NOTE: Must be in transaction
         logger.debug('Attempting to lock %r', self)
         with connections[using].cursor() as cursor:
-            cursor.execute("SELECT pg_try_advisory_xact_lock(%s::regclass::integer, %s);", (self._meta.db_table, self.id))
-            if not cursor.fetchone()[0]:
+            cursor.execute("SELECT pg_try_advisory_lock(%s::regclass::integer, %s);", (self._meta.db_table, self.id))
```

Comment:
Yes, testing became impossible without really heavy fixtures.

And this way we can lock something (SourceConfig) but still be submitting live changes to the DB (Log updates) with a single connection.